src/main.c

```diff
@@ -49,6 +52,61 @@ static void softReset_button_callback(struct device *port, struct gpio_callback
 	soft_reboot();
 }
 
+
+void cdc_acm_descriptor_callback (cdc_acm_cfg_t *cfg)
+{
+	customer_data_t *otp_data = (customer_data_t *)CUSTOMER_DATA_ADDRESS;
+
+	memset(cfg->vendor_string, '\0', 32);
+	memset(cfg->product_string, '\0', 32);
+	memset(cfg->serial_string, '\0', 32);
+
+	//read the USB descriptors from OTP and use some default values if it has not been properly populated
+	if((otp_data->vendor_name_len>0) && (otp_data->vendor_name_len<=32))
+	{
+		memcpy(cfg->vendor_string, otp_data->vendor_name, otp_data->vendor_name_len);
+	}
+	else
+	{
+		char *vendor = "Intel";
+		memcpy(cfg->vendor_string, vendor, strlen(vendor));
+	}
+
+	if((otp_data->board_name_len>0) && (otp_data->board_name_len<=32))
+	{
+		memcpy(cfg->product_string, otp_data->board_name, otp_data->board_name_len);
+	}
+	else
+	{
+		char *product =  "Arduino 101";
+		memcpy(cfg->product_string, product, strlen(product));
+	}
+
+	if((otp_data->product_sn_len>0) && (otp_data->product_sn_len<=32))
+	{
+		memcpy(cfg->serial_string, otp_data->product_sn, otp_data->product_sn_len);
+	}
+	else
+	{
+		char *serial = "00.01";
+		memcpy(cfg->serial_string, serial, strlen(serial));
+	}	
+
+
+	if((otp_data->product_vid != 0xFFFF) && (otp_data->product_vid != 0x0000)
```

Comment:
@eriknyquist 
If they really want 0000 or FFFF as the VID/PID for development purposes then I think they should just do it in code.
As a minimum I think we should check for FFFF because like @SidLeung  said that would indicate that the VID/PID has not been programmed.
Another option would be just comment the code and just leave an approrpiate comment